### ISSUE TITLE
[Question] How to write Unit tests

### ISSUE THREAD

**Issue by suho** [tquocit/survey-demo#4]
## Issue

Currently, there are no unit tests in the project. However, it's a requirement from the code challenge

![image](https://github.com/tquocit/survey-demo/assets/19943832/11787ee5-3e68-4a15-a91a-4c5d0554b724)

Could you share with me why you did not add any tests for the project?

If you will need to add tests, could you share with me how you will write tests?

**Comment by tquocit** [comment]
Yes. For this part, so sorry I don't have enough time to handle this part. To write unit test, I often use XCTest Framework from  Apple. Such as LoginViewController, we can test LoginButtonSuccess by creating mockUserManager, mockLoginViewModel. In mockLoginViewModel, we can create method checkTokenAndNavigater and showHomeScreen. So in methodTest we can check with XCTAssertTrue
To test with API, I'll use the URLProtocol class to handle with network and provide mock response

For almost projects and companies I joined are startup companies. So almost time we try to focus on implement features asap. Then we will comeback to write unit test for main features or APIs when have time.
